lib/dynflow/action/with_bulk_sub_plans.rb

```diff
@@ -0,0 +1,54 @@
+module Dynflow
+  module Action::WithBulkSubPlans
+    include Dynflow::Action::Cancellable
+
+    BATCH_SIZE=10
+
+    # Should return a slice of size items starting from item with index from
+    def entries(from, size)
+      raise NotImplementedError
+    end
+
+    def total_count
+      raise NotImplementedError
+    end
+
+    def current_batch
+      start_position = output[:total_count]
+      size = start_position + BATCH_SIZE > total_count ? total_count - start_position : BATCH_SIZE
+      entries(start_position, size)
+    end
+
+    def done?
+      super && total_count == output[:total_count]
+    end
+
+    def resume
+      # TODO
+      super
+    end
+
+    def run_progress
+      if counts_set? && output[:total_count] > 0
```

Comment:
The condition was there just to refrain from dividing by zero, this shouldn't happen anymore, removed it